### PR TITLE
[FIXED JENKINS-44864] Don't report deprecated setters if their values are redundant

### PR DESCRIPTION
**Pull request by stephenc**
See [JENKINS-44864](https://issues.jenkins-ci.org/browse/JENKINS-44864)

@reviewbybees

### CODE REVIEW

File: plugin/src/test/java/org/jenkinsci/plugins/structs/describable/DescribableModelTest.java [hunk 1]
```diff
@@ -873,4 +875,134 @@ public void testJavaStandardTypes() throws Exception {
...
\ No newline at end of file
+
+    @Test
+    public void given_model_when_fieldRenamed_then_uselessSettersIgnored() throws Exception {
+        EvolvedClass instance;
+        Map<String,Object> expected = new HashMap<String, Object>();
+
+        instance = new EvolvedClass(false);
+        expected.clear();
+        expected.put("option", Boolean.FALSE);
```
Comment (jglick) [inline plugin/src/test/java/org/jenkinsci/plugins/structs/describable/DescribableModelTest.java]: BTW `ImmutableMap.of` is probably the more readable way to write tests like these.

File: plugin/src/test/java/org/jenkinsci/plugins/structs/describable/DescribableModelTest.java [hunk 2]
```diff
@@ -873,4 +875,134 @@ public void testJavaStandardTypes() throws Exception {
                 "doubleValue1", 0.0
         ));
     }
-}
\ No newline at end of file
+
+    @Test
```
Comment (jglick) [inline plugin/src/test/java/org/jenkinsci/plugins/structs/describable/DescribableModelTest.java]: `@Issue`

**Comment by jglick** [comment]
(feel free to `deploy` a snapshot if you need this for downstream testing)

File: plugin/src/test/java/org/jenkinsci/plugins/structs/describable/DescribableModelTest.java (hunk 1 above)
Comment (stephenc) [inline plugin/src/test/java/org/jenkinsci/plugins/structs/describable/DescribableModelTest.java]: Well the issue with ImmutableMap.of is that you end up having to apply type hints everywhere, e.g. 

```
assertThat("Only required properties",
                DescribableModel.uninstantiate2_(instance).toMap(), Matchers<Map<String,Object>>.is(ImmutableMap.<String,Object>of("option", Boolean.FALSE)))
```

Comment (jglick) [inline plugin/src/test/java/org/jenkinsci/plugins/structs/describable/DescribableModelTest.java]: Not if you use `assertEquals`.

**Comment by stephenc** [comment]
@reviewbybees done

**Pull request merged**